pkg/controller/case_test.go

```diff
@@ -0,0 +1,583 @@
+		osb.LatestAPIVersion().HeaderValue(),
+		fakeRecorder,
+		7*24*time.Hour,
+		7*24*time.Hour,
+		"DefaultClusterIDConfigMapName",
+		"DefaultClusterIDConfigMapNamespace",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start and sync informers
+	testCase.stopCh = make(chan struct{})
+	informerFactory.Start(testCase.stopCh)
+	coreInformerFactory.Start(testCase.stopCh)
+	informerFactory.WaitForCacheSync(testCase.stopCh)
+	coreInformerFactory.WaitForCacheSync(testCase.stopCh)
+
+	// start the controller
+	go testController.Run(1, testCase.stopCh)
+
+	return testCase
+}
+
+func (ct *ControllerTest) TearDown() {
+	close(ct.stopCh)
+}
+
+func (ct *ControllerTest) AsyncForInstances() {
+	ct.fakeOSBClient.ProvisionReaction.(*fakeosb.ProvisionReaction).Response.Async = true
+	ct.fakeOSBClient.UpdateInstanceReaction.(*fakeosb.UpdateInstanceReaction).Response.Async = true
+	ct.fakeOSBClient.DeprovisionReaction.(*fakeosb.DeprovisionReaction).Response.Async = true
+}
+
+func (ct *ControllerTest) AsyncForBindings() {
+	ct.fakeOSBClient.BindReaction.(*fakeosb.BindReaction).Response.Async = true
+	ct.fakeOSBClient.UnbindReaction.(*fakeosb.UnbindReaction).Response.Async = true
+}
+
+func (ct *ControllerTest) AssertOSBBasicAuth(t *testing.T, username, password string) {
+	require.NotNil(t, ct.osbClientCfg, "OSB Client was not created, wait for broker is ready")
+	assert.Equal(t, ct.osbClientCfg.AuthConfig.BasicAuthConfig, &v2.BasicAuthConfig{
+		Username: username,
+		Password: password,
+	})
+}
+
+func (ct *ControllerTest) SetFirstOSBPollLastOperationReactionsInProgress(numberOfInProgressResponses int) {
+	numberOfPolls := 0
+	ct.fakeOSBClient.PollLastOperationReaction = fakeosb.DynamicPollLastOperationReaction(
+		func(_ *osb.LastOperationRequest) (*osb.LastOperationResponse, error) {
+			numberOfPolls++
+			state := osb.StateInProgress
+			if numberOfPolls > numberOfInProgressResponses {
+				state = osb.StateSucceeded
+			}
+			return &osb.LastOperationResponse{State: state}, nil
+		})
+}
+
+func (ct *ControllerTest) SetFirstOSBProvisionReactionsHTTPError(numberOfErrorResponses int) {
+	numberOfPolls := 0
+	ct.fakeOSBClient.ProvisionReaction = fakeosb.DynamicProvisionReaction(
+		func(_ *osb.ProvisionRequest) (*osb.ProvisionResponse, error) {
+			numberOfPolls++
+			if numberOfPolls > numberOfErrorResponses {
+				return &osb.ProvisionResponse{}, nil
+			}
+			return nil, osb.HTTPStatusCodeError{
+				StatusCode: http.StatusUnauthorized,
+			}
+		})
+}
+
+func (ct *ControllerTest) SetOSBBindReactionWithHTTPError(code int) {
+	ct.fakeOSBClient.BindReaction = &fakeosb.BindReaction{
+		Error: osb.HTTPStatusCodeError{
+			StatusCode: code,
+		},
+	}
+}
+
+func (ct *ControllerTest) spyOSBClientFunc(target v2.CreateFunc) v2.CreateFunc {
+	return func(osbCfg *v2.ClientConfiguration) (v2.Client, error) {
+		ct.osbClientCfg = osbCfg
+		return target(osbCfg)
+	}
+}
+
+func (ct *ControllerTest) createClusterServiceBroker() *v1beta1.ClusterServiceBroker {
```

Comment:
IMO is misleading:

you have function called `createClusterServiceBroker ` which is returning fixture and also `CreateSimpleClusterServiceBroker` which is calling k8s Create function. Maybe the first one should be called just `fixClusterServiceBroker`? U already using that for other func e.g. `fixCatalogResponse`

Reply:
renamedrenamed